### PR TITLE
Add a nix flake for tkn-watch

### PR DESCRIPTION
**Pull request by vdemeester**
This allow to run things tkn-watch like `nix run github:chmouel/tkn-watch --
--help`.
It also provides packages for other to integrate with their flake
configuration / nix derivation.

`nix develop` drops you in a development environment with everything
needed.
`nix flake check` runs all the tests (fmt, cargo test, …)

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by vdemeester** [comment]
```
error: build of '/nix/store/dkafxvfqzpjv1izcg7v57f9d3bndb9k1-tkn-watch-deps-0.3.0.drv' on 'ssh://builder@192.168.1.77' failed: error: builder for '/nix/store/dkafxvfqzpjv1izcg7v57f9d3bndb9k1-tkn-watch-deps-0.3.0.drv' failed with exit code 101;
              last 10 log lines:
              >     21:     0x55555557e661 - std::rt::lang_start::h7938b639184f4b2d
              >     22:     0x555555576663 - main
              >     23:     0x15555532f1d7 - __libc_start_call_main
              >     24:     0x15555532f297 - __libc_start_main_impl
              >     25:     0x555555572091 - _start
              >                                  at /build/glibc-2.34/csu/../sysdeps/x86_64/start.S:116
              >     26:                0x0 - <unknown>
              > warning: build failed, waiting for other jobs to finish...
              > error: build failed
              > [naersk] cargo returned with exit code 101, exiting
              For full logs, run 'nix log /nix/store/dkafxvfqzpjv1izcg7v57f9d3bndb9k1-tkn-watch-deps-0.3.0.drv'.
error: builder for '/nix/store/dkafxvfqzpjv1izcg7v57f9d3bndb9k1-tkn-watch-deps-0.3.0.drv' failed with exit code 1;
       last 10 log lines:
       >     21:     0x55555557e661 - std::rt::lang_start::h7938b639184f4b2d
       >     22:     0x555555576663 - main
       >     23:     0x15555532f1d7 - __libc_start_call_main
       >     24:     0x15555532f297 - __libc_start_main_impl
       >     25:     0x555555572091 - _start
       >                                  at /build/glibc-2.34/csu/../sysdeps/x86_64/start.S:116
       >     26:                0x0 - <unknown>
       > warning: build failed, waiting for other jobs to finish...
       > error: build failed
       > [naersk] cargo returned with exit code 101, exiting
       For full logs, run 'nix log /nix/store/dkafxvfqzpjv1izcg7v57f9d3bndb9k1-tkn-watch-deps-0.3.0.drv'.
error: 1 dependencies of derivation '/nix/store/3vpg2pksq49mbnj91zy1y5gs66nji9kj-tkn-watch-0.3.0.drv' failed to build
```

🤔

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    naersk.url = "github:nmattia/naersk";
+    naersk.inputs.nixpkgs.follows = "nixpkgs";
+  };
+
+  outputs = { self, nixpkgs, naersk }:
+    let
+      cargoToml = (builtins.fromTOML (builtins.readFile ./Cargo.toml));
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" ];
```
Comment (chmouel) [inline flake.nix:13]: i had issues with cross compilation on different host with the kube-rs lib (openssl) but i believe it's fixed so you can maybe add aarch64-darwin (if that's the name)

**Comment by chmouel** [comment]
can you try to "patch" the cargo.toml  here : 

https://github.com/chmouel/tkn-watch/blob/main/Cargo.toml#L26

removing the features = ["vendored"]

tldr; we need it for cross compile architecture between darwin arm64 and x86_64.

since we don't have a builder in github actions for arm64 we generate a universal binary from x86_64 with embedded bytecode of arm64 (that [rosetta](https://en.wikipedia.org/wiki/Rosetta_(software)) thing) and if we use the global libssl it would fail resolve on arm64, long story short we need to embed a compiled ssl version, maybe that's an issue for nix?

File: Cargo.toml [hunk 2]
```diff
@@ -23,7 +23,7 @@ chrono-humanize = "0.2.1"
 dialoguer = { version = "0.10.0", features = ["fuzzy-select"] }
 clap = { version = "3.1", features = ["derive", "cargo"] }
 regex = "1.5.5"
-openssl = { version = "0.10.38", features = ["vendored"] }
```
Comment (chmouel) [inline Cargo.toml:26]: this is needed for osx cross compile :\ 

no way to do this as a patch in nix ?

(but let's see if it works first)

Comment (vdemeester) [reply Cargo.toml:26]: There is ways to do it.. but in both case I can't compile it on Nix so far 🤔

Comment (chmouel) [reply Cargo.toml:26]: this is weird, do you have the full log by any chance?

**Comment by chmouel** [comment]
maybe this has the answer to our issue somewhere? https://johns.codes/blog/rust-enviorment-and-docker-build-with-nix-flakes

**Comment by vdemeester** [comment]
> maybe this has the answer to our issue somewhere? https://johns.codes/blog/rust-enviorment-and-docker-build-with-nix-flakes

Ah you are right, most likely I forgot to add `openssl` (and maybe others) as builtinput.. let's try.
